### PR TITLE
fix MarkDown formatting in autogenerated documentation

### PR DESCRIPTION
**Pull request by branfosj**
pulls in my suggested changes from https://github.com/easybuilders/easybuild-framework/pull/4117 and some other changes to solve the markdown lint issues from https://github.com/boegel/easybuild-docs/actions/runs/3837183470/jobs/6532198248

### CODE REVIEW

File: easybuild/tools/docs.py [hunk 1]
```diff
@@ -1261,7 +1259,7 @@ def gen_easyblocks_overview_md(package_name, path_to_examples, common_params=Non
     eb_links = []
     for eb_class in sorted(eb_classes, key=lambda c: c.__name__):
         eb_name = eb_class.__name__
-        eb_links.append("<a href='#" + eb_name.lower() + "'>" + eb_name + "</a>")
+        eb_links.append("[" + eb_name + "](" + eb_name.lower() + ")")
```
Comment (branfosj) [inline easybuild/tools/docs.py:1262]:
```suggestion
        eb_links.append("[" + eb_name + "](#" + eb_name.lower() + ")")
```

File: test/framework/docs.py [hunk 2]
```diff
@@ -526,7 +522,7 @@ def test_gen_easyblocks_overview(self):
                     self.assertTrue(name in ebdoc)
                     names.append(name)
 
-        toc = ["<a href='#" + n.lower() + "'>" + n + "</a>" for n in sorted(set(names))]
+        toc = ["\\[" + n + "\\]\\(" + n.lower() + "\\)" for n in sorted(set(names))]
```
Comment (branfosj) [inline test/framework/docs.py:525]:
```suggestion
        toc = ["\\[" + n + "\\]\\(#" + n.lower() + "\\)" for n in sorted(set(names))]
```

File: easybuild/tools/docs.py (hunk 1 above)
Comment (boegel) [inline easybuild/tools/docs.py:1262]: `#` is missing here to have a correct link to the subsections below:

```suggestion
        eb_links.append("[" + eb_name + "](#" + eb_name.lower() + ")")

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Pull request merged**